src/topology/README.md

```diff
@@ -0,0 +1,150 @@
+interface-topology
+========================
+
+> Implementation of the topology interface used by the `js-libp2p` registrar.
+
+This interface has two main purposes. It uniforms the registration of libp2p protocols and enables a smarter connection management.
+
+## Table of Contents
+
+- [Implementations](#implementations)
+- [Install](#install)
+- [Modules using the interface](#modulesUsingTheInterface)
+- [Usage](#usage)
+- [Api](#api)
+
+## Implementations
+
+### Topology
+
+A libp2p topology with a group of common peers.
+
+### Multicodec Topology
+
+A libp2p topology with a group of peers that support the same protocol.
+
+## Install
+
+```sh
+$ npm install libp2p-interfaces
+```
+
+## Modules using the interface
+
+TBA
+
+## Usage
+
+###  Topology
+
+```js
+const Topology = require('libp2p-interfaces/src/topology')
+
+const toplogy = new Topology({
+  min: 0,
+  max: 50
+})
+```
+
+### Multicodec Topology
+
+```js
+const MulticodecTopology = require('libp2p-interfaces/src/topology/multicodec-topology')
+
+const toplogy = new MulticodecTopology({
+  min: 0,
+  max: 50,
+  multicodecs: ['/echo/1.0.0'],
+  handlers: {
+    onConnect: (peerInfo, conn) => {},
+    onDisconnect: (peerInfo) => {}
+  }
+})
+
```

Comment:
I would just delete these 3 lines. We can explain the registrar property in the API section.
```suggestion
```